resources/requests.py

```diff
@@ -1,45 +1,168 @@
+import datetime
+
 from flask_restful import Resource, reqparse
 from flask import json, request, jsonify
-from models import dtrequest
+from resources.models import RequestModel, is_admin, find_by_username
 
 from flask_jwt_extended import (create_access_token,create_refresh_token,
 jwt_required, jwt_refresh_token_required,get_jwt_identity, get_raw_jwt)
 
+#set current user
+
+
+RM = RequestModel
+
+parser = reqparse.RequestParser()
+parser.add_argument('requestor',help='fill user id', required=True)
+parser.add_argument('request_type',help='please fill in request_type', required=True)
+parser.add_argument('status',help='request status', required=False)
+parser.add_argument('description',help='please fill in descritption', required=True)
+
 class RequestResource(Resource):
     
     """Get all request"""
     @jwt_required
     def get(self):    
-        #to set in utils
-        current_user = ["anto@gmail.com"]
-        res = [d for d in dtrequest if d["email"] in current_user]
-        return res
+        
+        current_user = get_jwt_identity()
+        user_id = find_by_username(current_user)[0]
+        check_admin = is_admin(current_user)
+        if  check_admin is True:
+            """returns all request"""
+            return RM.fetch_all_request()
+        return {current_user:RM.fetch_for_logged_in_user(user_id)}
 
-    """create new request"""
+    @jwt_required
     def post(self):
-        json_data = request.get_json(force=True)
-        if not json_data:
-            return {'message': 'No input data provided'}, 400
-        elif (len(json_data)!=len(dtrequest[0])):
-            return "Please enter all details"
-        dtrequest.append(json_data)
-        return {"status":"success", "data": json_data }, 201
+        """create new request"""
+        self.data = parser.parse_args()
+
+        requestor = self.data['requestor']
+        request_type = self.data['request_type']
+        status = self.data['status']
+        description = self.data['description']
+        created_on = datetime.datetime.now()
+
+        try:
+            save = RequestModel.create_request(self,requestor,request_type,status,description,created_on)
+            
+            return {"status":str(save), "data": request_type }, 201
+        
+        except:
+            return {"msg":"Failed to create new request"}
+        
+        #return {"status":"success", "data": self.data }, 201
         
     
     
 class Request(Resource):
     """Get request by ID"""
+    @jwt_required
     def get(self, req_id):
-        for _request in dtrequest:
-            if (req_id == _request['id']):
-                return _request, 200
-        return "request not found", 404
+        response = RM.fetch_request_by_id(self, req_id)[0][0]
+        current_user = get_jwt_identity()
+        user_id = find_by_username(current_user)[0]
+
+        if user_id == response['requestor']:
```

Comment:
have a response for a non-existent request instead of just returning an empty list.